### ISSUE TITLE
LT14: False positive on ORDER BY inside WITHIN GROUP clause (T-SQL STRING_AGG)

### ISSUE THREAD

**Issue by fank** [sqlfluff/sqlfluff#7546]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

Note: [#6495](https://github.com/sqlfluff/sqlfluff/issues/6495) reported the same issue for `OVER (ORDER BY ...)` window functions and was fixed in [PR #6909](https://github.com/sqlfluff/sqlfluff/pull/6909) by adding an exclusion config. However, the `WITHIN GROUP (ORDER BY ...)` case was mentioned in the comments but is **still not fixed** in 4.0.4.

### What Happened

LT14 flags the `ORDER BY` keyword inside a `WITHIN GROUP (ORDER BY ...)` clause as needing its own line. This is a false positive — `WITHIN GROUP (ORDER BY col)` is a single inline aggregate construct (used with `STRING_AGG` in T-SQL), not a top-level `ORDER BY` clause.

The same issue applies to `WHERE` inside inline subqueries like `NOT EXISTS (SELECT 1 FROM t WHERE ...)`.

### Expected Behaviour

The following query should pass LT14 without violations:

```sql
SELECT
    category_name
    ,product_list = STRING_AGG(product_name, ', ') WITHIN GROUP (ORDER BY product_name)
FROM
    dbo.products AS products
GROUP BY
    category_name
```

The `ORDER BY` inside `WITHIN GROUP (...)` is part of the aggregate function syntax, not a standalone clause.

### Observed Behaviour

```
L:   3 | P:  61 | LT14 | The 'ORDER' keyword should always start a new line.
                       | [layout.keyword_newline]
```

### How to reproduce

**test.sql:**
```sql
SELECT
    category_name
    ,product_list = STRING_AGG(product_name, ', ') WITHIN GROUP (ORDER BY product_name)
FROM
    dbo.products AS products
GROUP BY
    category_name
```

```bash
sqlfluff lint --rules LT14 --dialect tsql test.sql
```

Output:
```
== [test.sql] FAIL
L:   3 | P:  61 | LT14 | The 'ORDER' keyword should always start a new line.
                       | [layout.keyword_newline]
```

**Second example — WHERE inside NOT EXISTS:**

```sql
SELECT
    customer_name
FROM
    dbo.customers AS customers
WHERE
    NOT EXISTS (SELECT 1 FROM dbo.orders AS orders WHERE orders.customer_id = customers.id)
```

```bash
sqlfluff lint --rules LT14 --dialect tsql test2.sql
```

Flags the `WHERE` inside the `NOT EXISTS` subquery.

### Dialect

tsql

### Version

sqlfluff 4.0.4, Python 3.10.12

### Configuration

```ini
[sqlfluff]
dialect = tsql
templater = raw
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
I'll see if I can get something together to fix this. In the meantime, you can add this to your configuration to include the tsql `within_group_clause`.
```ini
[sqlfluff:layout:type:orderby_clause]
keyword_line_position_exclusions =
    window_specification,
    aggregate_order_by,
    withingroup_clause,
    within_group_clause,
    pipe_operator_clause
```

**Comment by fank** [comment]
Thanks @keraion — the `keyword_line_position_exclusions` workaround fixes the WITHIN GROUP case. We still see false positives on inline `WHERE` inside `NOT EXISTS` subqueries though (e.g. `NOT EXISTS (SELECT 1 FROM ... WHERE ...)`), so keeping this open for that.

**Comment by keraion** [comment]
I'd say the subquery might be a matter of preference, but this should work to not flag it.
```ini
[sqlfluff:layout:type:where_clause]
keyword_line_position_exclusions =
    bracketed
```

**Comment by peterbud** [comment]
* I *think* we could modify `OrderByClauseSegment` -> `AggregateOrderByClause` in `WithinGroupClause` to make it easier to differentiate between the two semantics to treat them differently
* At the same time, let me mention, that `ORDER BY` expression in `WITHIN GROUP` is semantically a plain `ORDER BY` clause, see [here](https://learn.microsoft.com/en-us/sql/t-sql/functions/string-agg-transact-sql?view=sql-server-ver17), so honestly I don't see any real reason why we should handle that differently - it's personal preference, and with `keyword_line_position_exclusions` you can treat it differently if you want.
* I don't see a ready-to-use solution for handling subqueries differently within a EXISTS/NOT EXISTS  vs other places. EXISTS is an operator with a subquery, and semantically they are indistinguishable from a simple SELECT statement (you can check the parsed structure). Here the best advice is to apply `-- noqa: LT14` for the given line if that's your preference.
